idaes/models/properties/modular_properties/base/generic_property.py

```diff
@@ -1341,80 +1352,86 @@ def initialize(
         # Create solver
         opt = get_solver(solver, optarg)
 
+        # ---------------------------------------------------------------------
+        # Initialize temperature_crit_mix and pressure_crit_mix
+        for k in blk.keys():
+            reference_phase = 'Liq'
```

Comment:
We shouldn't hard-code phase names like this. It's easy to see places down the road where we have separate aqueous phases and organic phases that are named as such.

Also, it looks like this method is run regardless of whether there are coexisting phases or not. What if the liquid and vapor phases have different equations of state that have different methods of calculating critical properties? I'm still of the opinion that critical properties should be tied to coexisting phase pairs.

Reply:
On the first point, I agree with @dallan-keylogic ; we should not hard-code names like that. To go a step further, there is nothing stopping someone from naming their liquid phase "foo" (an we explicitly support that).

On the second point, I am not sure that a phase pair is the correct way to go either, precisely for the point @dallan-keylogic  raised; what if vapour and liquid have different methods for critical properties (and potentially multiple liquid phases in future)? Rather, my initial feeling is that we need to state a specific phase for calculating the critical properties (i.e., pick vapor or liquid).

As an aside, there might be valid cases where a users wants to take a third option as well (i.e. choose a method for calculating the critical point using something other than the EoS/equivalent for the vapour or liquid phase). I don't think we want to support that for now, but identifying a specific phase would probably make it easier to add support for that in the future.

Reply:
@dallan-keylogic There are many ways we could do it. AS I was thinking of a response however I realised one good reason why a phase pair is not a good choice; it would only work for multi-phase systems, whilst there might be cases where you want to calculate critical properties for a single phase system.